### PR TITLE
kylin-3812 optimize the child CompareTupleFilter in a CompareTupleFilter

### PR DESCRIPTION
**Pull request by kyotoYaho**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java [hunk 1]
```diff
@@ -283,6 +289,20 @@ public TupleFilter acceptOptimizeTransformer(FilterOptimizeTransformer transform
...
+    private TupleFilter optimizeChildCompareTupleFilter(CompareTupleFilter child) {
+        FilterOptimizeTransformer transformer = new FilterOptimizeTransformer();
+        TupleFilter result = child.acceptOptimizeTransformer(transformer);
+        if (result == ConstantTupleFilter.TRUE) {
+            // use string instead of boolean since it's encoded as string
+            result = new ConstantTupleFilter("true");
+        } else if (result == ConstantTupleFilter.FALSE) {
+            result = new ConstantTupleFilter("false");
+        } else {
+            this.isNormal = false;
```
Comment (vaful) [inline core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java]: this means that TupleFilters excepted AlwaysTrue or AlwaysFalse will not pushdown?

File: core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java [hunk 2]
```diff
@@ -74,6 +77,9 @@ private CompareTupleFilter(CompareTupleFilter another) {
 
     @Override
     public void addChild(TupleFilter child) {
+        if (child instanceof CompareTupleFilter) {
```
Comment (vaful) [inline core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java]: plz add some test case in UT or IT

File: core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java (hunk 1 above)
Comment (kyotoYaho) [reply core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java]: Why do you think AlwaysTrue or AlwaysFalse cannot be pushed down? Current CompareTupleFilter does not well support the case that both children are CompareTupleFilter. Therefore, push down will be disabled. This behavior is not changed.

File: core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java (hunk 2 above)
Comment (kyotoYaho) [reply core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java]: We need to refine our IT and add more test cases. To test this case, it's necessary to add a column with boolean type.

**Pull request merged**